gui/src/2D/controls/grid.ts

```diff
@@ -137,7 +142,9 @@ export class Grid extends Container {
             return this;
         }
 
+        this._columnDefinitions[index].onChangedObservable.remove(this._columnDefinitionObservers[index]);
         this._columnDefinitions[index] = new ValueAndUnit(width, isPixel ? ValueAndUnit.UNITMODE_PIXEL : ValueAndUnit.UNITMODE_PERCENTAGE);
+        this._columnDefinitionObservers[index] = this._columnDefinitions[index].onChangedObservable.add(() => this._markAsDirty())!;

```

Comment:
you must ensure that this is also called during the dispose (I believe the row/col definitions are not cleaned during dispose for now)

Reply:
added a commit to clean up all resources.  Since child controls were already cleared, I believe cells should have been and then also row/column definitions!